shared/src/commonMain/kotlin/io/customer/shared/di/DIGraph.kt

```diff
@@ -0,0 +1,101 @@
+package io.customer.shared.di
+
+import kotlin.reflect.KClass
+
+/**
+ * Base dependency injection class to satisfy dependencies without adding any external dependency.
+ * Any DI component created within the SDK should inherit this class.
+ *
+ * @see [getNewInstance] and [getSingletonInstance] to get more details on how to use this class.
+ */
+abstract class DIGraph {
```

Comment:
I see value in creating DiGraph in KMM, but doing it this early makes me worried we are causing some major refactors. I don't see a reason to add it to KMM right now when we are currently working on just using KMM for background queue v2.

For now, I think we can keep DiGraph as-is in Android and iOS SDKs and each of those SDKs can add the background queue v2 instance as a dependency to their respective graphs.